### PR TITLE
Fix `_aws_get_role` to handle using the "default" profile

### PR DESCRIPTION
**Pull request by omus**
Fixes https://github.com/JuliaCloud/AWS.jl/issues/449. I found a few other changes that can be made here but as this is meant to be a bugfix I've minimized the changes to the "src" directory.

The config and credentials file used in the test was provided by @iamed2

### CODE REVIEW

**Comment by omus** [comment]
bors try

File: test/AWSCredentials.jl [hunk 1]
```diff
@@ -85,6 +85,25 @@ end
...
+
+    @testset "default profile" begin
+        access_key_id = "assumed_access_key_id"
+        config_dir = joinpath(@__DIR__, "configs", "default-role")
+
+        patch = Patches._assume_role_patch("AssumeRole"; access_key=access_key_id)
+
+        config = withenv(
+            "AWS_CONFIG_FILE" => joinpath(config_dir, "config"),
+            "AWS_SHARED_CREDENTIALS_FILE" => joinpath(config_dir, "credentials"),
```
Comment (omus) [inline test/AWSCredentials.jl:97]:
```suggestion
            "AWS_SHARED_CREDENTIALS_FILE" => joinpath(config_dir, "credentials"),
            "AWS_ACCESS_KEY_ID" => nothing,
            "AWS_SECRET_ACCESS_KEY => nothing,
```

File: test/AWSConfig.jl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    @testset "default profile assumes role" begin
+        access_key_id = "assumed_access_key_id"
+        config_dir = joinpath(@__DIR__, "configs", "default-role")
+
+        # Avoid calling out to STS with invalid credentials
+        patch = Patches._assume_role_patch("AssumeRole"; access_key=access_key_id)
+
+        config = withenv(
+            "AWS_CONFIG_FILE" => joinpath(config_dir, "config"),
+            "AWS_SHARED_CREDENTIALS_FILE" => joinpath(config_dir, "credentials"),
```
Comment (omus) [inline test/AWSConfig.jl:11]:
```suggestion
            "AWS_SHARED_CREDENTIALS_FILE" => joinpath(config_dir, "credentials"),
            "AWS_ACCESS_KEY_ID" => nothing,
            "AWS_SECRET_ACCESS_KEY => nothing,
```

File: test/AWSConfig.jl [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            "AWS_ACCESS_KEY_ID" => nothing,
+            "AWS_SECRET_ACCESS_KEY => nothing,
```
Comment (omus) [inline test/AWSConfig.jl:13]:
```suggestion
            "AWS_SECRET_ACCESS_KEY" => nothing,
```

File: test/AWSCredentials.jl [hunk 4]
```diff
@@ -85,6 +85,27 @@ end
...
+            "AWS_ACCESS_KEY_ID" => nothing,
+            "AWS_SECRET_ACCESS_KEY => nothing,
```
Comment (omus) [inline test/AWSCredentials.jl:99]:
```suggestion
            "AWS_SECRET_ACCESS_KEY" => nothing,
```

**Comment by omus** [comment]
bors try

File: test/AWSCredentials.jl [hunk 5]
```diff
@@ -85,6 +85,27 @@ end
...
         end
     end
+
+    @testset "default profile" begin
+        access_key_id = "assumed_access_key_id"
+        config_dir = joinpath(@__DIR__, "configs", "default-role")
+
+        patch = Patches._assume_role_patch("AssumeRole"; access_key=access_key_id)
+
+        config = withenv(
```
Comment (omus) [inline test/AWSCredentials.jl:95]:
```suggestion
        cred = withenv(
```

File: test/AWSCredentials.jl [hunk 6]
```diff
@@ -85,6 +85,27 @@ end
...
+            "AWS_ACCESS_KEY_ID" => nothing,
+            "AWS_SECRET_ACCESS_KEY" => nothing,
+        ) do
+            ini = read(Inifile(), ENV["AWS_CONFIG_FILE"])
+            apply(patch) do
+                AWS._aws_get_role("default", ini)
+            end
+        end
+
+        @test config.credentials.access_key_id == access_key_id
```
Comment (omus) [inline test/AWSCredentials.jl:107]:
```suggestion
        @test cred.access_key_id == access_key_id
```

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
I'll merge this tomorrow if no one objects or doesn't merge it first

File: test/configs/default-role/credentials [hunk 7]
```diff
@@ -0,0 +1,3 @@
+[test]
+aws_access_key_id = foo
+aws_secret_access_key = bar
```
Comment (mattBrzezinski) [inline test/configs/default-role/credentials:3]:
```suggestion
aws_secret_access_key = bar

```

**Review by mattBrzezinski — APPROVED** [review]
bors r+

**Pull request merged**